lib/util/git/index.ts

```diff
@@ -348,10 +330,19 @@ export async function getCommitMessages(): Promise<string[]> {
 
 export async function setBranch(branchName: string): Promise<CommitSha> {
   await syncGit();
-  if (!(await branchExists(branchName))) {
+  if (!branchExists(branchName)) {
     throwBranchValidationError(branchName);
   }
   logger.debug(`Setting current branch to ${branchName}`);
+  if (!branchName.startsWith(config.branchPrefix)) {
```

Comment:
This logic was previously in branchExists() so it was kind of just "lucky" that branchExists() was always called by the application prior to setBranch()